### PR TITLE
docs: Update RELEASES md file (#311)

### PR DESCRIPTION
**Pull request by Oleksandr123234**
# Pull Request Template

## Description
This change contains updates for RELEASES.md file.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Pull Request contains one commit. I squashed my commits

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
 </details>
 
+## Version 3.10.0 <a name="3.10.0"></a> (October 8, 2024)
+
+## What's New
+
+These release notes start with introducing Bitbucket Git server support. This feature enriches the users' toolset and makes the platform more attractive for Bitbucket users.
+
+As a mean of adapting to the uproar the revolves around Artificial Intelligence, we equip KubeRocketCI with a [chat assistant](https://docs.kuberocketci.io/docs/next/user-guide/add-ai-assistant) support feature.
+By integrating chat assistants with the platform, users can get an intelligent assistant which can provide them with informational support.
```
Comment (NikolayMarusenko) [inline RELEASES.md:68]: assistant can provide any support, it depend on assistant configuration

File: RELEASES.md [hunk 2]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
+
+A new look comes with a new release. Meet our new [documentation site](https://docs.kuberocketci.io/). This site was reworked in a way to match the style of KubeRocketCI portal.
+Besides, the new version boasts documentation versioning and improved navigation.
+
+To improve user experience and clarity in your CI/CD flows, we significantly redesigned the Environment creation procedure. To start from, we updated terminology within the platform.
+From now on, Environments become Deployment Flows and Stages are now Environments. This renaming was done for better reflecting the purpose of each entity.
+Additionally, Deployment Flows and Environments are now created separately and have new settings to apply configurable pipelines.
+
+Regarding of pipelines, this release also presents major enhancements to pipelines. To address users' specific needs, we are working on making pipelines more customizable.
+Now users can make up their own build, review, or deploy pipelines and easily apply them to their when both creating and editing Codebases and Environments.
```
Comment (NikolayMarusenko) [inline RELEASES.md:78]: 'Now users can make up their own build, review, or deploy pipelines' - previously we also have this feature, this feature more about flexible and easy approach how to define pipelines and customization

File: RELEASES.md [hunk 3]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
+Apart from that, we introduce a new pipeline type called "clean". This pipeline is launched when clicking the new "Clean" button within Environments and performs [customizable application deletion](https://docs.kuberocketci.io/docs/next/operator-guide/cd/customize-environment-deletion) logic.
```
Comment (NikolayMarusenko) [inline RELEASES.md:79]: It is not about pipeline, it is about flow of deletion.

File: RELEASES.md [hunk 4]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
+
+Another crucial enhancement is a new section called Pipelines. This section aims to improve pipeline observability process by gathering all the pipelines in a single place.
+
+It is also worth mentioning that we are continuously working on widening the platform capabilities, this time we announce that KubeRocketCI is now capable of deploying RPM packages in addition to traditional Helm charts. Better still, KubeRocketCI now natively supports C and C++ languages with Make and CMake build tools.
```
Comment (NikolayMarusenko) [inline RELEASES.md:83]: 'traditional' - rephrase please

File: RELEASES.md [hunk 5]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
+* In this release, KubeRocketCI can now deploy RPM packages apart from common Helm charts. ([#272](https://github.com/epam/edp-tekton/issues/272))
+* Ansible tool support has been added to the platform. In the context of the platform, this platform will serve as a mean of deploying RPM packages. ([#311](https://github.com/epam/edp-headlamp/issues/311))
+* When creating codebases, users can now select Deployment Option. It can be either Helm chart or RPM package. ([#363](https://github.com/epam/edp-headlamp/issues/363))
+* The Make and CMake application pipelines have been added as build tools for C and C++ languages. ([#260](https://github.com/epam/edp-tekton/issues/260))
+* Add codebase codemie integration ([#323](https://github.com/epam/edp-headlamp/issues/323))
+* A new tab called "Variables" has been added to the Environment (formerly Stage) page. In this tab, users can set key-value variables to put into the application as environment variables. ([#404](https://github.com/epam/edp-headlamp/issues/404))
```
Comment (NikolayMarusenko) [inline RELEASES.md:109]: this is major feature, lets highlight in top of doc

File: RELEASES.md [hunk 6]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
 * [Version 2.6.0](#2.6.0)
 </details>
 
+## Version 3.10.0 <a name="3.10.0"></a> (October 8, 2024)
```
Comment (NikolayMarusenko) [inline RELEASES.md:61]: we have major feature with autodeploy, please provide details for this case

File: RELEASES.md [hunk 7]
```diff
@@ -57,6 +58,109 @@ Get acquainted with the latest EDP releases.
...
+
+## What's New
+
+These release notes start with introducing Bitbucket Git server support. This feature enriches the users' toolset and makes the platform more attractive for Bitbucket users.
```
Comment (NikolayMarusenko) [inline RELEASES.md:65]: please rephrase that we support new VCS provider

File: RELEASES.md [hunk 8]
```diff
@@ -57,6 +58,110 @@ Get acquainted with the latest EDP releases.
...
+
+Additionally, we have added new functionality to our deployment pipelines. A new tab called Variables has been added to the Environment (formerly Stage) details page. In this tab, users can specify key-value parameters, which will be transmitted within the application as environment variables. This improvement allows users to configure application deployment settings without leaving the KubeRocketCI portal.
+
+Deploy pipelines now support manual approval. When creating a custom deploy pipeline, users can include a checkpoint to prompt for approval or rejection before proceeding to the next step.
+
+We have added a new deployment trigger type called "[Auto-stable](https://github.com/epam/edp-cd-pipeline-operator/blob/master/deploy-templates/crds/v2.edp.epam.com_stages.yaml#L161)" has been added to the platform. This trigger type allows for creating an ordered queue of deploy Pipeline Runs, which will be launched sequentially once the previous one finishes. Unlike the "Auto" trigger type, where intermediate pipelines are skipped, "Auto-stable" ensures that each version is deployed, making it easier to identify the exact application version that caused errors during deployment.
```
Comment (SergK) [inline RELEASES.md:85]: deployment strategy

File: RELEASES.md [hunk 9]
```diff
@@ -57,6 +58,110 @@ Get acquainted with the latest EDP releases.
...
+
+We have also introduced the ability to [customize Environment deletion](https://docs.kuberocketci.io/docs/next/operator-guide/cd/customize-environment-deletion). This feature is particularly useful for users with applications that have specific configurations or dependencies, e.g. database cleanup, infrastructure tearing down, etc. Customized clean flows make it easier to maintain components throughout their lifecycle.
+
+Additionally, we have added new functionality to our deployment pipelines. A new tab called Variables has been added to the Environment (formerly Stage) details page. In this tab, users can specify key-value parameters, which will be transmitted within the application as environment variables. This improvement allows users to configure application deployment settings without leaving the KubeRocketCI portal.
+
+Deploy pipelines now support manual approval. When creating a custom deploy pipeline, users can include a checkpoint to prompt for approval or rejection before proceeding to the next step.
```
Comment (SergK) [inline RELEASES.md:83]: Deployment pipelines have been enhanced with a manual approval feature. This allows developers to integrate a checkpoint within their custom deployment pipelines, requiring a manual review and approval process before the pipeline can advance to the next stage. This ensures greater control and oversight over the deployment process.

File: RELEASES.md [hunk 10]
```diff
@@ -57,6 +58,110 @@ Get acquainted with the latest EDP releases.
...
+
+We have added the ability to specify custom build and review pipelines with just one click when editing or creating new codebase branches. This enhancement increases flexibility when working with codebases, allowing for easier modifications to pipeline logic based on specific requirements.
+
+Another significant enhancement is the introduction of a new section called Pipelines. This section centralizes pipeline observability by consolidating all pipelines in one place. We are continuing to migrate Tekton functionality into our platform. Consequently, we have consolidated our pipelines into a single location. The new Pipelines section displays the entire Tekton Pipeline Run history, simplifying pipeline management and observability.
+
+Users can now deliver their applications in the form of RPM packages, implemented using Ansible. Correspondingly, support for C and C++ languages with make and CMake build tools has been added to the platform.
```
Comment (SergK) [inline RELEASES.md:77]: let's split this and:

Users can now automate the deployment of their applications packaged as RPMs, utilizing Ansible for this process.

File: RELEASES.md (hunk 8 above)
Comment (SergK) [reply RELEASES.md:85]: original: where intermediate pipelines are skipped

smth like: where intermediate versions are skipped and the latest versions are deployed

**Review by SergK — CHANGES_REQUESTED** [review]

File: RELEASES.md [hunk 11]
```diff
@@ -57,6 +58,112 @@ Get acquainted with the latest EDP releases.
 * [Version 2.6.0](#2.6.0)
 </details>
 
+## Version 3.10.0 <a name="3.10.0"></a> (October 11, 2024)
+
+## What's New
+
+We are enhancing the platform towards new code storage and Version Control Systems support. We are pleased to introduce Bitbucket Cloud as a new Version Control System in KubeRocketCI.
+
+Our platform now integrates with Large Language Models, such as ChatGPT, as an intelligent [chat assistant](https://docs.kuberocketci.io/docs/next/user-guide/add-ai-assistant). Depending on your specified settings, it can perform a variety of tasks. We can also set up this assistant for you. Please contact our team for more details.
```
Comment (SergK) [inline RELEASES.md:67]: `Depending on your specified settings`

Depending on your assistant's settings

File: RELEASES.md [hunk 12]
```diff
@@ -57,6 +58,112 @@ Get acquainted with the latest EDP releases.
...
+
+We have added the ability to specify custom build and review pipelines with just one click when editing or creating new codebase branches. This enhancement increases flexibility when working with codebases, allowing for easier modifications to pipeline logic based on specific requirements.
+
+Another significant enhancement is the introduction of a new section called Pipelines. This section centralizes pipeline observability by consolidating all pipelines in one place. We are continuing to migrate Tekton functionality into our platform. Consequently, we have consolidated our pipelines into a single location. The new Pipelines section displays the entire Tekton Pipeline Run history, simplifying pipeline management and observability.
+
+We are expanding the options for managing applications. Users are now able to automate the deployment of their applications packaged as RPMs, utilizing Ansible for this process.
```
Comment (SergK) [inline RELEASES.md:77]: `We are expanding the options for managing applications.`

We are expanding the options for delivering applications.

File: RELEASES.md [hunk 13]
```diff
@@ -57,6 +58,112 @@ Get acquainted with the latest EDP releases.
...
+
+As part of our expanding language support, we’ve added C and C++ to our CI/CD platform. This includes integration with the make and CMake build tools, enhancing the platform's capabilities for building and managing C/C++ projects.
+
+We have also introduced the ability to [customize Environment deletion](https://docs.kuberocketci.io/docs/next/operator-guide/cd/customize-environment-deletion). This feature is particularly useful for users with applications that have specific configurations or dependencies, e.g. database cleanup, infrastructure tearing down, etc. Customized clean flows make it easier to maintain components throughout their lifecycle.
+
+Additionally, we have added new functionality to our deployment pipelines. A new tab called Variables has been added to the Environment (formerly Stage) details page. In this tab, users can specify key-value parameters, which will be transmitted within the application as environment variables. This improvement allows users to configure application deployment settings without leaving the KubeRocketCI portal.
```
Comment (SergK) [inline RELEASES.md:83]: This release adds new functionality to deployment pipelines. A new tab, "Variables," has been added to the Environment details page (previously known as the Stage details page). In this tab, users can specify key-value pairs that will be injected into the deployment pipeline as environment variables. This enhancement allows users to configure application deployment settings directly within the KubeRocketCI portal, providing greater flexibility and control over the deployment process.

File: RELEASES.md [hunk 14]
```diff
@@ -57,6 +58,112 @@ Get acquainted with the latest EDP releases.
...
+
+Deployment pipelines have been enhanced with a manual approval feature. This allows developers to integrate a checkpoint within their custom deployment pipelines, requiring a manual review and approval process before the pipeline can advance to the next stage. This ensures greater control and oversight over the deployment process.
```
Comment (SergK) [inline RELEASES.md:85]: We are pleased to introduce a new Manual Approval feature on the platform. This feature allows developers to incorporate an approval task within their custom deployment pipelines, requiring a manual review and approval before advancing to the next stage. While deployment pipelines are a primary use case for this capability, the approval task can also be leveraged in other workflows. This enhancement provides greater control and oversight, ensuring all processes meet required standards and criteria before progressing.

File: RELEASES.md [hunk 15]
```diff
@@ -57,6 +58,112 @@ Get acquainted with the latest EDP releases.
...
+
+Our platform now integrates with Large Language Models, such as ChatGPT, as an intelligent [chat assistant](https://docs.kuberocketci.io/docs/next/user-guide/add-ai-assistant). Depending on your assistant's settings, it can perform a variety of tasks. We can also set up this assistant for you. Please contact our team for more details.
+
+A new release brings a new look. Check out our new [documentation site](https://docs.kuberocketci.io/), which features improved navigation and documentation versioning support.
+
+We have updated terminology within the platform. Environments are now referred to as Deployment Flows, and Stages are now Environments. These changes align our terminology with globally recognized terms, reducing user confusion. For more details, please refer to our updated [Glossary](https://docs.kuberocketci.io/docs/next/glossary).
+
+We have added the ability to specify custom build and review pipelines with just one click when editing or creating new codebase branches. This enhancement increases flexibility when working with codebases, allowing for easier modifications to pipeline logic based on specific requirements.
```
Comment (SergK) [inline RELEASES.md:73]: let's remove 'with just one click'

**Review by SergK — APPROVED** [review]

**Pull request merged**